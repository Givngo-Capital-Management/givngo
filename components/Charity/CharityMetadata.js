import React from 'react';
import { Button, Divider, Icon } from 'semantic-ui-react';
import './CharityMetadata.scss';

export default function CharityMetadata() {
  return (
    <div className="charity-metadata">
      <h3>Video title</h3>
      <div className="charity-stats">
        <span>Mar 19, 2020</span>
        <div className="charity-actions">
          <Button basic icon labelPosition="left">
            <Icon name="share" />
            Share
          </Button>
        </div>
      </div>
      <Divider />
    </div>
  );
}
