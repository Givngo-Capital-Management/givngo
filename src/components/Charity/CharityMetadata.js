import React from 'react';
import { Button, Divider, Icon } from 'semantic-ui-react';
import styles from './CharityMetadata.module.scss';
import moment from 'moment';

export default function CharityMetadata(props) {
  const { charity } = props;
  const date = moment.utc(charity.video_date * 1000).format('MMMM DD, YYYY').toString()
  return (
    <div className={styles.metadata}>
      <h3>{charity.video_title}</h3>
      <div className={styles.stats}>
        <span>{date}</span>
        <div className={styles.actions}>
          <Button basic icon labelPosition="left">
            <Icon name="share" />
            Share
          </Button>
        </div>
      </div>
      <Divider />
    </div>
  );
}
