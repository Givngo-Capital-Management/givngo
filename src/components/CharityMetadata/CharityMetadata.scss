@import '../../styles/shared.scss';

.charity-metadata {
  color: #707070;
  font-size: 16px;

  h3 {
    font-weight: 400;
    color: $text-color-dark;
  }

  .charity-stats {
    display: flex;
    justify-content: space-between;
    width: 100%;
    align-items: center;

    .charity-actions {
      display: flex;
      align-items: center;
      & > *:not(:last-child) {
        margin-right: 8px;
      }
    }
  }
}