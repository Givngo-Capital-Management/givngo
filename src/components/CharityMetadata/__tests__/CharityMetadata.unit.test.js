import React from 'react';
import { shallow } from 'enzyme';
import CharityMetadata from '../CharityMetadata';

describe('CharityMetadata', () => {
  test('renders', () => {
    const wrapper = shallow(<CharityMetadata />);
    expect(wrapper).toMatchSnapshot();
  });
});
