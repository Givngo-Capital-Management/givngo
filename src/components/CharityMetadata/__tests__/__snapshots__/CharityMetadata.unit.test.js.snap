// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharityMetadata renders 1`] = `
<div
  className="charity-metadata"
>
  <h3>
    Video title
  </h3>
  <div
    className="charity-stats"
  >
    <span>
      Mar 19, 2020
    </span>
    <div
      className="charity-actions"
    >
      <Button
        as="button"
        basic={true}
        icon={true}
        labelPosition="left"
      >
        <Icon
          as="i"
          name="share"
        />
        Share
      </Button>
    </div>
  </div>
  <Divider />
</div>
`;
